dojo/api_v2/serializers.py

```diff
@@ -1046,247 +1041,55 @@ class ImportScanSerializer(serializers.Serializer):
         endpoint_to_add = data['endpoint_to_add']
         scan_date = data['scan_date']
-        scan_date_time = datetime.datetime.combine(scan_date, timezone.now().time())
-        if settings.USE_TZ:
-            scan_date_time = timezone.make_aware(scan_date_time, timezone.get_default_timezone())
-
         # Will save in the provided environment or in the `Development` one if absent
-        environment_name = data.get('environment', 'Development')
         version = data.get('version', None)
         build_id = data.get('build_id', None)
         branch_tag = data.get('branch_tag', None)
         commit_hash = data.get('commit_hash', None)
 
+        environment_name = data.get('environment', 'Development')
         environment = Development_Environment.objects.get(name=environment_name)
         tags = None
         if 'tags' in data:
             logger.debug('import scan tags: %s', data['tags'])
             tags = data['tags']
 
-        print(tags)
-        test = Test(
-            engagement=data['engagement'],
-            lead=data['lead'],
-            test_type=test_type,
-            target_start=data['scan_date'],
-            target_end=data['scan_date'],
-            environment=environment,
-            percent_complete=100,
-            version=version,
-            branch_tag=branch_tag,
-            build_id=build_id,
-            commit_hash=commit_hash,
-            tags=tags)
+        engagement = data['engagement']
+        lead = data['lead']
+
+        scan = data.get('file', None)
+        endpoints_to_add = [endpoint_to_add] if endpoint_to_add else None
+
+        importer = Importer()
         try:
-            test.full_clean()
-        except ValidationError:
-            pass
+            test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, lead, environment,
+                                                                             active=active, verified=verified, tags=tags,
+                                                                             minimum_severity=minimum_severity,
+                                                                             endpoints_to_add=endpoints_to_add,
+                                                                             scan_date=scan_date, version=version,
+                                                                             branch_tag=branch_tag, build_id=build_id,
+                                                                             commit_hash=commit_hash,
+                                                                             push_to_jira=push_to_jira,
+                                                                             close_old_findings=close_old_findings)
+        # convert to exception otherwise django rest framework will swallow them as 400 error
+        # exceptions are already logged in the importer
+        except SyntaxError as se:
+            raise Exception(se)
```

Comment:
could we use ValidationError or another one instead of untyped exception?

Reply:
those will be swallowed by the rest framework and turned into 400 errors without any info.